wifiphisher/extensions/deauth.py

```diff
@@ -88,101 +62,116 @@ def get_packet(self, packet):
-            else:
-                bssid = self._data.target_ap_bssid
-                channels.append(self._data.target_ap_channel)
-
-            # create a list of addresses that are not acceptable
-            non_valid_list = self._non_client_addresses +\
-                self._observed_clients
-
-            # if sender or receirver is valid and not already a
-            # discovered client check to see if either one is a client
-            if receiver not in non_valid_list and sender not in non_valid_list:
-
-                # in case the receiver is the access point
-                if receiver == bssid:
-                    # add the sender to the client list
-                    self._observed_clients.append(sender)
-
-                    # create and add deauthentication packets for client
-                    self.packets_to_send += self._craft_packet(sender,
-                                                               receiver,
-                                                               bssid)
-                    self.packets_to_send += self._craft_packet(receiver,
-                                                               sender,
-                                                               bssid)
-
-                # in case the sender is the access point
-                elif sender == bssid:
-                    # add the receiver to the client list
-                    self._observed_clients.append(receiver)
-
-                    # create and add deauthentication packets for client
-                    self.packets_to_send += self._craft_packet(receiver,
-                                                               sender,
-                                                               bssid)
-                    self.packets_to_send += self._craft_packet(sender,
-                                                               receiver,
-                                                               bssid)
-        return (channels, self.packets_to_send)
+        channels = list()
+        packets_to_send = list()
+
+        # when --essid is used we don't have target_ap_bssid
+        if self._data.target_ap_bssid:
+            # we need to create this packet only once
+            if self._first_run:
+                # Craft and add deauth/disas packet to broadcast address
+                packets_to_send += self._craft_packet(self._data.target_ap_bssid,
+                                                      constants.WIFI_BROADCAST,
+                                                      self._data.target_ap_bssid)
+                self._first_run = False
+
+            bssid = self._data.target_ap_bssid
+            channels.append(self._data.target_ap_channel)
+        else:
+            bssid = packet.addr3
+
+            # Do not send deauth for our rogue access point
+            if bssid == self._data.rogue_ap_mac:
+                return ([], [])
```

Comment:
`self._data.rogue_ap_mac` will always equal to `None` since we have set it to `None` in `pywifiphsiher.py`. #698 will fix this.

@blackHatMonkey do you think we can merge #698 first and then I can do a full functional test for #700 ?